pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -972,6 +977,7 @@ func (l *LibvirtDomainManager) migrateHelper(vmi *v1.VirtualMachineInstance, opt
 	if err != nil {
 		return fmt.Errorf("error encountered during MigrateToURI3 libvirt api call: %v", err)
 	}
+	close(migrationOver)
```

Comment:
I think we should `defer` this close putting it at the beginning of this function.
The `migrateHelper` early returns an error if any occurs.
Then in the `(l *LibvirtDomainManager) migrate` the error is sent to the `migrationErrorChan` channel and it is read by the monitor.
In this flow, the `migrationOver` channel is never closed.
Also, deferring the `close(migrationOver)` allow us to immediately stop the monitor because instead of waiting for 400ms
```
		select {
		case <-m.migrationOver:
		case <-time.After(monitorSleepPeriodMS * time.Millisecond):
		}
```
Does that sound like reasoning to you? Thanks

Reply:
@fossedihelm Great catch! when `migrateHelper` return an error we don't need to wait 400ms as well. 
Maybe we can even move that part of monitor creation and launch into the `migrateHelper` itself. Then we can 
create the migration over channel inside the migrateHelper itself, so that the channel management will be encapsulated inside the migrateHelper.

Reply:
Good catch indeed! Thanks for the reviews, PTAL!